app/models/mixins/tenancy_mixin.rb

```diff
@@ -13,4 +17,16 @@ def accessible_tenant_ids(user_or_group, strategy)
       tenant.accessible_tenant_ids(strategy)
     end
   end
+
+  def set_tenant
+    # Priority
+    #   Owning group
+    #   Current user
+    #   Parent EMS
+    #   Root tenant
+    self.tenant_id ||= try(:miq_group).try(:tenant_id)             ||
+                       User.current_tenant.try(:id)                ||
+                       try(:ext_management_system).try(:tenant_id) ||
+                       Tenant.root_tenant.try(:id)
```

Comment:
"Do or do not, there is no try."  Yoda wouldn't be proud of all of these conditionals "hidden" behind try calls so let's see if we can find a better way but this should hopefully fix many of the failures on travis.  cc @gtanzillo